vulcan/lib/client/jsx/contexts/input_state_management.tsx

```diff
@@ -77,28 +78,80 @@ export function WithBufferedInputs({
     }
   }, [cancelInputs, commitSessionInputChanges, stepName]);
 
+  function setAutoPass(event: any, checked: boolean) {
+    if (checked) {
+      dispatch(setStepUIAutoPass(stepName))
+    } else {
+      dispatch(clearStepUIAutoPass(stepName))
+    }
+  }
+
+  function isPassable(stepName: string | null) {
+    if (stepName!=null) {
+      // ToDo: better way?
+      const this_step = state.workflow?.steps[0].filter((val) => val.name == stepName)[0]
+      if (this_step && this_step.doc!=null && this_step.doc.startsWith('SKIPPABLE')) return true
```

Comment:
Ultimate first trigger of 'Can this step be auto-passed?'.  We could do this part differently, but what I like about it is that it operates within the bounds of CWL, falls to the workflow author to set up / maintain, and is visible to workflow users behind the (?).

Reply:
Seems interesting, I don't mind.  One q: does this mean that primary inputs are not "skippable"?

Reply:
Well... not by these exact methods. Wasn't thinking towards that use case.

I imagine the downstream `triggerCommit` and `triggerRun` machinery could be utilized for primary input auto-skipping too, but we'd need a starting point that's not reliant on the user marking them for auto_complete... likely just an 'if_CWL_fetaure' then mark start out new (completely fresh) figures of that workflow with `triggerCommit: ['primary_inputs']`.  And this exact 'doc' method for marking in the CWL won't work because each individual primary input has it's own 'doc'-section... there's not a single 'doc' for all primary inputs which could be started with "SKIPPABLE".